uxarray/grid.py

```diff
@@ -453,28 +451,41 @@ def integrate(self, var_ds, quadrature_rule="triangular", order=4):
 
         return integral
 
-    def _build_edge_node_connectivity(self):
+    def _build_edge_node_connectivity(self, repopulate=False):
         """Constructs the UGRID connectivity variable (``Mesh2_edge_nodes``)
         and stores it within the internal (``Grid.ds``) and through the
         attribute (``Grid.Mesh2_edge_nodes``).
 
         Additionally, the attributes (``inverse_indices``) and
         (``fill_value_mask``) are stored for constructing other
         connectivity variables.
+
+        Parameters
+        ----------
+        repopulate : bool, optional
+            Flag used to indicate if we want to overwrite the existed `Mesh2_edge_nodes` and generate a new
+            inverse_indices, default is False
         """
-        padded_face_nodes = close_face_nodes(self.Mesh2_face_nodes.values,
-                                             self.nMesh2_face,
-                                             self.nMaxMesh2_face_nodes)
 
-        # array of empty edge nodes where each entry is a pair of indices
-        edge_nodes = np.empty((self.nMesh2_face * self.nMaxMesh2_face_nodes, 2),
-                              dtype=INT_DTYPE)
+        # need to derive edge nodes
+        if "Mesh2_edge_nodes" not in self.ds or repopulate:
+            padded_face_nodes = close_face_nodes(self.Mesh2_face_nodes.values,
+                                                 self.nMesh2_face,
+                                                 self.nMaxMesh2_face_nodes)
 
-        # first index includes starting node up to non-padded value
-        edge_nodes[:, 0] = padded_face_nodes[:, :-1].ravel()
+            # array of empty edge nodes where each entry is a pair of indices
+            edge_nodes = np.empty(
+                (self.nMesh2_face * self.nMaxMesh2_face_nodes, 2),
+                dtype=INT_DTYPE)
 
-        # second index includes second node up to padded value
-        edge_nodes[:, 1] = padded_face_nodes[:, 1:].ravel()
+            # first index includes starting node up to non-padded value
+            edge_nodes[:, 0] = padded_face_nodes[:, :-1].ravel()
+
+            # second index includes second node up to padded value
+            edge_nodes[:, 1] = padded_face_nodes[:, 1:].ravel()
+        else:
+            # If "Mesh2_edge_nodes" already exists, directly return the function call
```

Comment:
The reason why I did this is because it actually did the exactly same thing as the previous implementation. See my comments below about why I deleted the redundant "mpas" test case